client/app/actions/commentsActionCreators.js

```diff
@@ -43,24 +45,34 @@ export function submitCommentFailure(error) {
 
 export function fetchComments() {
   return dispatch => {
+    dispatch(setIsFetching());
     return (
-      commentsManager
-        .fetchComments()
-        .then(res => dispatch(fetchCommentsSuccess(res.data)))
-        .catch(res => dispatch(fetchCommentsFailure(res.data)))
+        commentsManager
+            .fetchComments()
+            .then(res => dispatch(fetchCommentsSuccess(res.data)))
+            .catch(res => dispatch(fetchCommentsFailure(res.data)))
     );
   };
 }
 
 export function submitComment(comment) {
+  const startTime = (new Date()).getTime();
   return dispatch => {
-    dispatch(incrementAjaxCounter());
+    dispatch(setIsSaving());
     return (
       commentsManager
         .submitComment(comment)
-        .then(res => dispatch(submitCommentSuccess(res.data)))
+        .then(res => {
+          const result = res.data;
+          const successTime = (new Date()).getTime();
+          const timeDiff = successTime - startTime;
+          if (timeDiff < minimumDelay) {
+            setTimeout(dispatch, minimumDelay - timeDiff, submitCommentSuccess(result));
+          } else {
+            dispatch(submitCommentSuccess(result));
+          }
+        })
         .catch(res => dispatch(submitCommentFailure(res.data)))
```

Comment:
There's no need for middleware if we're just going put back the sleep in the server. The middleware was for the idea of having the action start a timeout for the processing indicator. I agree with you that it's just too much to get into having this extra code in the submit action. It's already complicated enough with the thunk middleware to do the asynch save.